OnlineShop/OnlineShop.DAL/Repositories/Implementations/UserRepository.cs

```diff
@@ -1,17 +1,47 @@
-﻿using OnlineShop.DAL.Entities.Implementations;
+﻿using Microsoft.EntityFrameworkCore;
+using OnlineShop.DAL.Entities.Implementations;
+using OnlineShop.DAL.Infrastructure;
 using OnlineShop.DAL.Repositories.Interfaces;
 
 namespace OnlineShop.DAL.Repositories.Implementations;
 
-public class UserRepository:IBaseRepository<User>
+public class UserRepository(ShopDbContext dbContext) : IUserRepository
 {
-    public Task<User> GetById(Guid id)
+    public async Task<User> GetById(Guid id)
```

Comment:
Always add the suffix "Async" to the names of methods that have an Async or async modifie